{
  "origin": "codeshovel",
  "repositoryName": "hadoop",
  "repositoryPath": "/home/shaiful/research/codeshovel/codeshovel-projects/hadoop/.git",
  "startCommitName": "HEAD",
  "sourceFileName": "CallableSupplier.java",
  "functionName": "waitForCompletion",
  "functionId": "waitForCompletion___future-CompletableFuture__T__(modifiers-final)",
  "sourceFilePath": "hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/CallableSupplier.java",
  "functionStartLine": 114,
  "functionEndLine": 125,
  "numCommitsSeen": 2,
  "timeTaken": 1425,
  "changeHistory": [
    "e02eb24e0a9139418120027b694492e0738df20a"
  ],
  "changeHistoryShort": {
    "e02eb24e0a9139418120027b694492e0738df20a": "Yintroduced"
  },
  "changeHistoryDetails": {
    "e02eb24e0a9139418120027b694492e0738df20a": {
      "type": "Yintroduced",
      "commitMessage": "HADOOP-15183. S3Guard store becomes inconsistent after partial failure of rename.\n\nContributed by Steve Loughran.\n\nChange-Id: I825b0bc36be960475d2d259b1cdab45ae1bb78eb\n",
      "commitDate": "20/06/19 1:56 AM",
      "commitName": "e02eb24e0a9139418120027b694492e0738df20a",
      "commitAuthor": "Steve Loughran",
      "diff": "@@ -0,0 +1,12 @@\n+  public static \u003cT\u003e void waitForCompletion(\n+      final CompletableFuture\u003cT\u003e future)\n+      throws IOException {\n+    try (DurationInfo ignore \u003d\n+            new DurationInfo(LOG, false, \"Waiting for task completion\")) {\n+      future.join();\n+    } catch (CancellationException e) {\n+      throw new IOException(e);\n+    } catch (CompletionException e) {\n+      raiseInnerCause(e);\n+    }\n+  }\n\\ No newline at end of file\n",
      "actualSource": "  public static \u003cT\u003e void waitForCompletion(\n      final CompletableFuture\u003cT\u003e future)\n      throws IOException {\n    try (DurationInfo ignore \u003d\n            new DurationInfo(LOG, false, \"Waiting for task completion\")) {\n      future.join();\n    } catch (CancellationException e) {\n      throw new IOException(e);\n    } catch (CompletionException e) {\n      raiseInnerCause(e);\n    }\n  }",
      "path": "hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/CallableSupplier.java"
    }
  }
}